wallet/chainntfns.go

```diff
@@ -160,7 +160,14 @@ func (w *Wallet) ChainSwitch(ctx context.Context, forest *SidechainForest, chain
 			}
 		}
 
-		for _, n := range chain {
+		birthdayState := w.txStore.BirthState(dbtx)
+		var birthdayTime time.Time
+		if birthdayState != nil && birthdayState.SetFromTime {
+			// Use a day before the set birthday just to be safe.
+			birthdayTime = birthdayState.Time.Add(time.Hour * -24)
```

Comment:
This should be handled at the input, not here. If I restore, and set `2020-01-01 12:00:00` as birthday time, then you _can_ use that, as I (the user) have explicitly set the birthday in the past and am sure of it (and that's the exact value I intend to use).

What is _not_ safe is using `time.Now()` for a newly created wallet, because a new block may come in with an older timestamp than `time.Now()`.

Reply:
Adding verification to the input area and removing here.